xinference/web/ui/src/components/themeSwitch.js

```diff
@@ -0,0 +1,52 @@
+import { styled } from '@mui/material'
+import Switch from '@mui/material/Switch';
+
+const ThemeSwitch = styled(Switch)(({ theme }) => ({
+  "width": 80,
+  "height": 38,
+  "padding": 0,
+  '& .MuiSwitch-switchBase': {
+    "margin": 1,
+    "padding": 0,
+    "transform": 'translateX(5px) translateY(3px)',
+    '&.Mui-checked': {
+      "color": '#fff',
+      "transform": 'translateX(45px) translateY(3px)',
+      '& .MuiSwitch-thumb:before': {
+        backgroundImage: `url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" height="20" width="20" viewBox="0 0 20 20"><path fill="${encodeURIComponent(
```

Comment:
What happens when requesting `http://www.w3.org/2000/svg` in an environment without network？Should we use this svg as a static resource?